Move Kotlin extensions to separate module

kpavlov opened the pull request:
## Issue
Move Kotlin extensions to a separate module

## Change
- Introduce `langchain4j-kotlin` module
- 🔥 **Breaking change: move kotlin extensions under `dev.langchain4j.kotlin.*`**
- Update documentation


## General checklist
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ] There are no breaking changes
- [ ] I have added unit and/or integration tests for my change
- [ ] The tests cover both positive and negative cases
- [x] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [x] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green
<!-- Before adding documentation and example(s) (below), please wait until the PR is reviewed and approved. -->
- [x] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)


## Checklist for adding new maven module
<!-- Please double-check the following points and mark them like this: [X] -->
- [x] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`


## Checklist for adding new embedding store integration
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

kpavlov committed:
Add Kotlin module and dependencies for LangChain4j

Introduce a new `langchain4j-kotlin` module with its POM setup, including dependencies for Kotlin standard library and coroutines. Update parent, BOM, and project configurations to integrate the new module and manage exclusions for transitive dependencies.

kpavlov committed:
Merge branch 'main' into kotlin-module

kpavlov committed:
Move kotlin extensions from langchain4j and langchain4j-core to langchain4j-kotlin module

kpavlov committed:
Merge branch 'main' into kotlin-module

kpavlov committed:
Move kotlin extensions under dev.langchain4j.kotlin

kpavlov committed:
Update documentation

kpavlov committed:
Merge branch 'main' into kotlin-module

kpavlov committed:
Update documentation

dliubarskyi reviewed (commented):
@kpavlov thanks a lot!

Some Kotlin dependencies, profiles, tests have left in the `langchain4j` and `langchain4j-core` modules, could you please check?

kpavlov commented:
There are a couple of Kotlin tests in langchain4j-core testing Java classes. Let's keep them. I will clean up production dependencies. Thank you

kpavlov committed:
Merge branch 'main' into kotlin-module

kpavlov committed:
Refactor tests to replace assertk with Kotest matchers

Updated test assertions to use Kotest matchers for collections, maps, and types instead of assertk.

kpavlov committed:
Remove unused Kotlin dependencies and adjust Maven plugins

kpavlov committed:
Merge remote-tracking branch 'origin/kotlin-module' into kotlin-module

dliubarskyi reviewed (approved):
@kpavlov thanks a lot!
